src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/HoverEndpoint.cs

```diff
@@ -56,33 +61,81 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
             positionInfo.LanguageKind));
     }
 
-    protected override Task<VSInternalHover?> TryHandleAsync(TextDocumentPositionParams request, RazorRequestContext requestContext, DocumentPositionInfo positionInfo, CancellationToken cancellationToken)
+    protected override async Task<VSInternalHover?> TryHandleAsync(TextDocumentPositionParams request, RazorRequestContext requestContext, DocumentPositionInfo positionInfo, CancellationToken cancellationToken)
     {
         var documentContext = requestContext.DocumentContext;
         if (documentContext is null)
         {
-            return SpecializedTasks.Null<VSInternalHover>();
+            return null;
+        }
+
+        // HTML can still sometimes be handled by razor. For example hovering over
+        // a component tag like <Counter /> will still be in an html context
+        if (positionInfo.LanguageKind == RazorLanguageKind.CSharp)
+        {
+            return null;
         }
 
-        return _hoverService.GetRazorHoverInfoAsync(
-                documentContext,
-                positionInfo,
-                request.Position,
-                cancellationToken);
+        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+
+        // Sometimes what looks like a html attribute can actually map to C#, in which case its better to let Roslyn try to handle this.
+        // We can only do this if we're in single server mode though, otherwise we won't be delegating to Roslyn at all
```

Comment:
There is no check for SingleServer, and it seems there wasn't in HoverService before, but I think that is probably a regression from when the service was introduced.

I think this needs a check for single server added back. The comment is correct as far as I can tell

Reply:
Thanks! I'll take a look.

Reply:
Yes, you're right that this regressed at exactly [the moment](https://github.com/dotnet/razor/commit/a8e386f7e40ea661ac35e544edcd9f466c4a83ed) you predicted! It looks like the `SingleServerSupport` check was just accidentally dropped. I'll add it back. Great catch!